libc/main.c

```diff
@@ -39,13 +39,80 @@ void runner(void)
 	RUN_TEST_GROUP(unistd_file);
 	RUN_TEST_GROUP(wchar_wcscmp);
 	RUN_TEST_GROUP(test_pthread_cond);
+}
+
+
+/* crete directory unless it exists */
+static int libc_createDirIfMissing(const char *path)
+{
+	struct stat buffer;
 
-	unsetenv("POSIXLY_CORRECT");
+	if (stat(path, &buffer) != 0) {
+		if (errno == ENOENT) {
+			if (mkdir(path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) != 0) {
+				fprintf(stderr, "Creating %s directory by mkdir failed: %s\n", path, strerror(errno));
+				return -1;
+			}
+		}
+		else {
+			fprintf(stderr, "stat() on %s directory failed: %s\n", path, strerror(errno));
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+/* crete file with optional data (may be NULL) unless it exists */
+static int libc_createFileIfMissing(const char *path, const char *charData)
```

Comment:
IMHO naming functions which are not part of a libc - `libc_something` is really misleading